### PR TITLE
fix(flows): fix CI, remove old parsing

### PR DESCRIPTION
**Pull request by bodinsamuel**
## Describe your changes

- **Fix CI for Flows**
New flows format from the other repo is breaking the CI test here (inconvenient but that's it)

- **Finally, drop `nango-config.service.js`**
I realized the CI was broken because the types were not synchronized in the tests, when I tried to fix the types it was still incompatible elsewhere so decided to go all in and finally delete the old parser. 

- **Use `nango-yaml `parser to parse `flows.yaml`**
It's good because there is only one way to parse yaml now BUT:
The types are now slightly wrong in the other way. I still need to move `StandardNangoConfig` to `@nangohq/types` and change the `model` shape to the new shape (the one from nango-yaml parser). 
It's not an issue because this data is only used by the frontend that expects both type of models.

- **Cache `flows.yaml` parsing**

### CODE REVIEW

File: packages/shared/lib/services/flow.service.ts [hunk 1]
```diff
@@ -3,65 +3,121 @@ import path from 'path';
...
         try {
             const flowPath = path.join(dirname(import.meta.url), '../../flows.yaml');
-            const flows = yaml.load(fs.readFileSync(flowPath).toString()) as Config;
+            const flows = yaml.load(fs.readFileSync(flowPath).toString()) as FlowsYaml;
 
             if (flows === undefined || !('integrations' in flows) || Object.keys(flows.integrations).length <= 0) {
                 throw new Error('empty_flows');
             }
 
+            this.flowsRaw = flows;
```
Comment (TBonnin) [inline packages/shared/lib/services/flow.service.ts]: why not assigning to this.flowsRaw above directly?

File: packages/shared/lib/services/flow.service.ts [hunk 2]
```diff
@@ -3,65 +3,121 @@ import path from 'path';
...
+                continue;
+            }
 
-            const { success, response } = loadStandardConfig(nangoConfig);
+            const { models, ...rest } = flow;
+            const parser = new NangoYamlParserV2({
+                raw: { integrations: { [providerConfigKey]: rest }, models: models },
+                yaml: ''
+            });
+            parser.parse(); // we assume it's valid
```
Comment (TBonnin) [inline packages/shared/lib/services/flow.service.ts]: why can we assume it is valid?

File: packages/shared/lib/services/flow.service.ts [hunk 3]
```diff
@@ -3,65 +3,121 @@ import path from 'path';
...
+                providerConfigKey,
+                actions: [],
+                syncs: []
+            };
 
-            if (success && response) {
-                if (rawName) {
-                    const responseWithRaw = response.map((standardConfigItem) => {
-                        return { ...standardConfigItem, rawName };
+            for (const item of [...integration.actions, ...integration.syncs]) {
```
Comment (TBonnin) [inline packages/shared/lib/services/flow.service.ts:68]: can integration be undefined if not found above?

**Review by TBonnin — APPROVED** [review]

File: packages/shared/lib/services/flow.service.ts (hunk 2 above)
Comment (bodinsamuel) [reply packages/shared/lib/services/flow.service.ts]: it's been parsed already by the integration template repo, the parsing is there just to get a standard format

File: packages/shared/lib/services/flow.service.ts (hunk 3 above)
Comment (bodinsamuel) [reply packages/shared/lib/services/flow.service.ts:68]: in theory, everything is possible, in practice no. This file has plenty of issues regarding error handling, we'll need to go over it one more time for sure

File: packages/shared/lib/services/flow.service.ts (hunk 1 above)
Comment (bodinsamuel) [reply packages/shared/lib/services/flow.service.ts]: I can indeed 👍🏻

**Pull request merged**